hbase-server/src/test/java/org/apache/hadoop/hbase/executor/TestExecutorService.java

```diff
@@ -231,9 +231,11 @@ public void testSnapshotHandlers() throws Exception {
     executorService.startExecutorService(ExecutorType.MASTER_SNAPSHOT_OPERATIONS, 1);
 
     CountDownLatch latch = new CountDownLatch(1);
+    final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
I think you can simplify this with another CountDownLatch instead of busy polling. Something like follows.

    CountDownLatch latch = new CountDownLatch(1);
    CountDownLatch waitForEventToStart = new CountDownLatch(1);
    executorService.submit(new EventHandler(server, EventType.C_M_SNAPSHOT_TABLE) {
      @Override
      public void process() throws IOException {
        waitForEventToStart.countDown();
        try {
          latch.await();
        } catch (InterruptedException e) {
          Thread.currentThread().interrupt();
        }
      }
    });
    waitForEventToStart.await(10, TimeUnit.SECONDS);  <== Add a timeout as you wish.
    int activeCount = executorService.getExecutor(ExecutorType.MASTER_SNAPSHOT_OPERATIONS)
        .getThreadPoolExecutor().getActiveCount();
    Assert.assertEquals(activeCount, 1);
    latch.countDown();

Doesn't probably matter that much because this is just a test.

Reply:
Thanks for you advice